### ISSUE TITLE
adding custom chugins

### ISSUE THREAD

**Issue by cchafe** [ccrma/webchuck-ide#37]
I drag a .chug.wasm into the file list and it sets the correct serverFilename and an
identical virtualFilename. Is it the case that it actually needs different entries in order to 
function as a chugin? My working examples (edited by hand) of exported pages had to have the virtualFilename pre-pended with '/chugins/'

**Comment by terryzfeng** [comment]
Yes you can manually load them into the `/chugins` folder or use the call `loadChugin` function before `init` for that to be handled automatically.

**Comment by cchafe** [comment]
Any way to do this in the IDE? Dragging in a file wouldn't be enough.

**Comment by terryzfeng** [comment]
There's no way to load chugins into the IDE because chugins cannot be loaded after the VM starts (which is immediate on page load). @gewang might still be working on runtime chugin loading with some `@import` type syntax...

The preloaded list of chugins could be hackily modified if you wanted other chugins to be loaded:
https://github.com/ccrma/webchuck-ide/blob/main/src/utils/webChugins.ts 
https://github.com/ccrma/webchuck-ide/blob/6c9a9656e600e7827f923378f2136f03d779dba1/src/host.ts#L66C1-L68C67

**Comment by terryzfeng** [comment]
@cchafe @gewang Our preliminary testing of chuck v1.5.3.3 (unreleased) with the new `@import` syntax has allowed us to add custom chugins into the IDE! If you put the IDE into dev mode, you should be able to try this out! Let me know if you get a chance to test it and more details coming soon!

**Comment by cchafe** [comment]
very cool -- tested with examples/import/import-test-1.ck
I suppose the project exporter might be using an earlier chuck
///////////////////////////
<compiled.code>:2:1: syntax error [Chuck.js:834:16](https://cdn.jsdelivr.net/npm/webchuck@1.2.8/dist/Chuck.js)
[2] @import "Foo.ck

**Comment by terryzfeng** [comment]
Yes, In File > Settings, you can select use Development branch of ChucK (1.5.3.3-dev). This should allow you to use the new import syntax

**Comment by terryzfeng** [comment]
One thing to note is that you must *upload* a .ck file (only those can get imported). Examples that are loaded from the library can't be imported unfortunately. A bit of a fake filesystem hack in the IDE atm..

**Comment by terryzfeng** [comment]
The latest [webchuck](https://github.com/ccrma/webchuck/releases/tag/v1.2.10) package updates ChucK's wasm to 1.5.4.4, which enables the uploading of chugins to the IDE. This issue will be closed as done!
